### PR TITLE
PYMT-1479: Elasticsearch Health Check

### PR DESCRIPTION
**Pull request by brendonofficial**
This PR implements the Elasticsearch health check in externals for use in Payments and Subscriptions.

**Note:** Please do not update the branch with master. It is aligned with the `v1.1.0` tag. _Also_, please do not move the Jira ticket to Done.

### CODE REVIEW

File: src/Health/Checks/ElasticsearchHealthCheck.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        $this->client = $client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check(): HealthState
+    {
+        try {
+            $this->client->getIndices();
```
Comment (merk) [inline src/Health/Checks/ElasticsearchHealthCheck.php]: Lets move to using this API:

https://www.elastic.co/guide/en/elasticsearch/reference/6.8/cluster-stats.html

You should check that the "status" key is yellow or green. If no response or red, degraded

Comment (brendonofficial) [reply src/Health/Checks/ElasticsearchHealthCheck.php]: There is a PR for this implementation here: https://github.com/eonx-com/search/pull/28

File: composer.json [hunk 2]
```diff
@@ -57,7 +58,13 @@
       "Tests\\EoneoPay\\Externals\\": "tests/"
     }
   },
+  "extra": {
```
Comment (sjdaws) [inline composer.json]: I think this should be removed now that composer.lock is up to date

Comment (brendonofficial) [reply composer.json]: Removing this drops the `eonx-com/search` version back down to `2.1.7`, so this will have to remain in place.

**Pull request merged**